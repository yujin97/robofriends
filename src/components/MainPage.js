import React, { Component }from 'react';
import CardList from './CardList';
import Scroll from './Scroll';
import SearchBox from './SearchBox';
import ErrorBoundry from './ErrorBoundry';
import Header from './Header';
import './MainPage.css';

class MainPage extends Component {
	componentDidMount() {
		this.props.onRequestRobots();
	}

    filterRobots = () => {
        return this.props.robots.filter( robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        })
    }

	render() {
		const {  onSearchChange, isPending } = this.props;

		return isPending?
			<h1>Loading</h1>:
			(
			<div className = 'tc'>
				<h1 className = 'f2'> RoboFriends</h1>
				<Header />
				<SearchBox searchChange = {onSearchChange} />
				<Scroll>
					<ErrorBoundry>
						<CardList robots = {this.filterRobots()}/>
					</ErrorBoundry>
				</Scroll>
			</div>
			);
		}
}

export default MainPage;