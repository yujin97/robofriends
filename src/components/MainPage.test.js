import { shallow } from 'enzyme';
import React from 'react'
import MainPage from './MainPage'

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false,
    }
    wrapper = shallow(<MainPage {...mockProps}/>)
})

it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filter robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            isPending: false,
        }],
        searchField: 'john',
        isPending: false,
    }
    let wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filterRobots()).toEqual([
        {
            id: 3,
            name: 'John',
            isPending: false,
        }
    ]);
})

it('filter robots correctly 2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            isPending: false,
        }],
        searchField: 'a',
        isPending: false,
    }
    const filteredRobots = [];
    let wrapper2 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper2.instance().filterRobots()).toEqual(filteredRobots);
})