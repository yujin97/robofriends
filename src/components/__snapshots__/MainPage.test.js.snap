// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <h1
    className="f2"
  >
     RoboFriends
  </h1>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
